src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5830,6 +5845,28 @@ private static BoundExpression CreatePlaceholderIfNecessary(BoundExpression expr
 
             return null;
 
+            // Gets the instance or extension invocation receiver if any.
+            bool tryGetReceiver(BoundCall node, [MaybeNullWhen(returnValue: false)] out BoundCall receiver)
+            {
+                if (node.ReceiverOpt is BoundCall instanceReceiver)
+                {
+                    receiver = instanceReceiver;
+                    return true;
+                }
+
+                if (node.InvokedAsExtensionMethod && node.Arguments is [BoundCall extensionReceiver, ..] &&
+                    // Exclude arguments that need saving state before visiting (only lambdas currently).
+                    !VisitArgumentEvaluatePrologue(extensionReceiver).HasValue)
```

Comment:
>VisitArgumentEvaluatePrologue [](http://example.com/codeflow?start=21&length=29)

It looks like we are calling this helper twice for the same lambda argument. This doesn't quite fit into a typical pattern around methods with the given name:
- Prologue done once
- main work done once
- Epilogue done once

Reply:
> It looks like we are calling this helper twice for the same lambda argument

I'm not sure why would that be the case. I think `tryGetReceiver` should be called only once for each `node`. But nevertheless, this is fixed by implementing the suggestion from your other comment.

Reply:
>I'm not sure why would that be the case. I think ```tryGetReceiver``` should be called only once for each node.

It is true that ```tryGetReceiver``` is called only once for each node. However, the helper is called here and then it is called in ```VisitArgumentEvaluate``` when we don't enter this ```if```.